### PR TITLE
split `ControlDeck` between `Ship::` and `LUS::`

### LINKED ISSUE #706: split `ControlDeck` between ship and LUS
anything that touches

https://github.com/Kenix3/libultraship/blob/00c6bcc88286173ce22b8a6656373d3aa5830c62/src/controller/controldeck/ControlDeck.h#L35-L36

should move to the LUS version that inherits from the ship version

### PR DESCRIPTION
**Pull request by briaguya0**
resolves https://github.com/Kenix3/libultraship/issues/706

soh https://github.com/HarbourMasters/Shipwright/pull/4687

### CODE REVIEW

File: src/Context.cpp [hunk 1]
```diff
@@ -218,7 +218,7 @@ void Context::InitControlDeck(std::vector<CONTROLLERBUTTONS_T> additionalBitmask
         return;
     }
 
-    mControlDeck = std::make_shared<ControlDeck>(additionalBitmasks);
+    mControlDeck = std::make_shared<LUS::ControlDeck>(additionalBitmasks);
```
Comment (briaguya0) [inline src/Context.cpp]: reason this is in draft

Comment (Kenix3) [reply src/Context.cpp]: Hmm. What is the additional bitmasks doing again?

We could push this to the derived class. If we do that, we want some kind of internal overload able init function that's call from the main one.

Comment (briaguya0) [reply src/Context.cpp]: we don't have a derived context

Comment (Kenix3) [reply src/Context.cpp]: I meant for the ControlDeck class.

Comment (briaguya0) [reply src/Context.cpp]: even without `additionalBitmasks` we can't make an instance of an abstract class
```
error: invalid new-expression of abstract class type ‘Ship::ControlDeck’
```

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**